### PR TITLE
fixed duedate phpdoc

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: src/Issue/IssueField.php [hunk 1]
```diff
@@ -443,7 +443,7 @@ public function setSecurityId($id)
     /**
      * set issue's due date.
      *
-     * @param \DateTimeInterface|null $duedate due date string or DateTimeInterface object
```
Comment (staabm) [inline src/Issue/IssueField.php]: the implementation of this methods also allows to pass in a string

File: src/Issue/IssueField.php [hunk 2]
```diff
@@ -92,7 +92,7 @@ class IssueField implements \JsonSerializable
     /** @var string|null */
     public $resolutiondate;
 
-    /** @var \DateTimeInterface|null */
+    /** @var \DateTimeInterface|string|null */
```
Comment (staabm) [inline src/Issue/IssueField.php]:
```suggestion
    /** @var string|null */
```

**Comment by staabm** [comment]
hmm not sure whether the testerrors are related

**Comment by lesstif** [comment]
hi @staabm .

PHP json mapper will be process to ''\DateTimeInterface|string' is concrete class name, so it's throw class not found exception.

mapping annotation syntax are single or double type name with | delimiter for mapping annotation, but you are write the three type name.

thanks!

**Comment by staabm** [comment]
> PHP json mapper will be process to ''\DateTimeInterface|string' is concrete class name, so it's throw class not found exception.

sorry, I cannot follow.

> mapping annotation syntax are single or double type name with | delimiter, but you are write the three type name.

phpdoc allow any number of types separated by `|`.

as the implementation of `setDueDate` looks like, the internal property will be set to a string (and this conflicts with the documented phpdoc)

**Comment by lesstif** [comment]
ok, you are right, PHPdoc allow any number of types.

but this repos depends on PHP jsonmapper third party library then does not allow multiple property types mapping like this (\DateTimeInterface|string|null).

**Pull request closed without merging**